commands/show-bl-hostname.go

```diff
@@ -155,6 +156,10 @@ func getBlacklistedHostnameResultsView(res *resources.Resources, sort string, li
 		}},
 	}
 
+	if !noLimit {
```

Comment:
The limit should still go immediately after the `$sort` step for efficiency reasons.
> When a $sort precedes a $limit and there are no intervening stages that modify the number of documents, the optimizer can coalesce the $limit into the $sort. This allows the $sort operation to only maintain the top n results as it progresses, where n is the specified limit, and ensures that MongoDB only needs to store n items in memory. This optimization still applies when allowDiskUse is true and the n items exceed the aggregation memory limit.
https://docs.mongodb.com/manual/reference/operator/aggregation/limit/

You'll probably have to break up the query and append the `$project` stage afterwards.